### PR TITLE
Issue #30: Auto Save on Theme Change

### PR DESCRIPTION
**Pull request by orenmendelow**
Selecting a different theme within club settings will trigger a save and a page reload, allowing users to seamlessly shuffle between themes.

https://github.com/user-attachments/assets/6dea8926-6cb1-4b64-8f45-3fec5df183f5

## Summary by CodeRabbit

- **New Features**
	- Introduced Turbo Streams for enhanced interactivity in club updates, improving response handling for both successful and unsuccessful updates.
	- Added a new Stimulus controller for automatic form submission and page refresh upon changes to the theme selection.
  
- **Bug Fixes**
	- Enhanced error handling for scenarios where no questions are available or when a club is not found. 

- **Style**
	- Updated form rendering logic to include a data attribute for the theme field, improving integration with the new Stimulus controller.

### CODE REVIEW

File: app/controllers/clubs_controller.rb [hunk 1]
```diff
@@ -61,11 +61,10 @@ def update
     respond_to do |format|
       if @club.update(club_params)
         format.html { redirect_to edit_club_url(@club), notice: "Club was successfully updated." }
-        format.json { render :show, status: :ok, location: @club }
+        format.turbo_stream { head :ok }
       else
-        flash.now[:alert] = @club.errors.full_messages.join("<br/>")
         format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @club.errors, status: :unprocessable_entity }
+        format.turbo_stream { render turbo_stream: turbo_stream.replace("content", partial: "form", locals: { club: @club }) }
```
Comment (yosefserkez) [inline app/controllers/clubs_controller.rb:67]: You dont need to delete format.json. You can just add format.turbo_stream no?

File: app/javascript/controllers/theme_controller.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+export default class extends Controller {
+  connect() {
+    this.element.addEventListener('change', this.submit.bind(this))
+  }
+
+  submit() {
+    this.element.form.requestSubmit()
+    setTimeout(() => window.location.reload(), 100)
+  }
```
[the remark this answers is not in the document]
Comment (orenmendelow) [reply app/javascript/controllers/theme_controller.js:11]: None of these proposed changes work.

File: app/javascript/controllers/theme_controller.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    this.element.addEventListener('change', this.submit.bind(this))
+  }
```
[the remark this answers is not in the document]
Comment (orenmendelow) [reply app/javascript/controllers/theme_controller.js:6]: Done.

File: app/views/clubs/_form.html.erb [hunk 4]
```diff
@@ -20,7 +20,7 @@
           <%= render_sub_options(form, b.object.last[:sub_options]) if b.object.last[:sub_options].present? %>
         <% end %>
       <% elsif [:select, :collection_select, :time_zone_select].include?(option[:field_type]) %>
-        <%= form.send(option[:field_type], key, option[:options], {}, {}) %>
+        <%= form.send(option[:field_type], key, option[:options], {}, { data: { controller: "theme" } }) if key == :theme %>
```
[the remark this answers is not in the document]
Comment (orenmendelow) [reply app/views/clubs/_form.html.erb:23]: The Turbo Stream method didn't work for me, and removing the setTimeout and page reload therefore is necessary for this functionality.

File: app/controllers/clubs_controller.rb [hunk 5]
```diff
@@ -61,11 +61,10 @@ def update
     respond_to do |format|
       if @club.update(club_params)
         format.html { redirect_to edit_club_url(@club), notice: "Club was successfully updated." }
-        format.json { render :show, status: :ok, location: @club }
+        format.turbo_stream { head :ok }
```
[the remark this answers is not in the document]
Comment (orenmendelow) [reply app/controllers/clubs_controller.rb:64]: Not sure what this did but done.

File: app/controllers/clubs_controller.rb (hunk 1 above)
Comment (orenmendelow) [reply app/controllers/clubs_controller.rb:68]: Right, fixed.

**Comment by yosefserkez** [comment]
Closed in favor of https://github.com/yosefserkez/mailpals/commit/0af500cc8c03806d660920ccae3b464fbf0f817d

**Pull request closed without merging**